packages/apps/human-app/frontend/src/modules/worker/hooks/use-register-address.ts

```diff
@@ -1,73 +1,107 @@
 import { useMutation, useQueryClient } from '@tanstack/react-query';
-import { z } from 'zod';
 import { t } from 'i18next';
-import { useAuthenticatedUser } from '@/modules/auth/hooks/use-authenticated-user';
+import { z } from 'zod';
 import { apiClient } from '@/api/api-client';
 import { apiPaths } from '@/api/api-paths';
+import { useAuthenticatedUser } from '@/modules/auth/hooks/use-authenticated-user';
+import { useAccessTokenRefresh } from '@/api/hooks/use-access-token-refresh';
 import {
   PrepareSignatureType,
   prepareSignature,
 } from '@/api/hooks/use-prepare-signature';
 import type { ResponseError } from '@/shared/types/global.type';
-import { useAccessTokenRefresh } from '@/api/hooks/use-access-token-refresh';
 import { useWalletConnect } from '@/shared/contexts/wallet-connect';
 
-const RegisterAddressSuccessSchema = z.unknown();
+interface RegisterAddressCallbacks {
+  onSuccess?: () => void | Promise<void>;
+  onError?: (error: ResponseError) => void | Promise<void>;
+}
+
+interface RegisterAddressParams {
+  address: string;
+  signature: string;
+}
+
+interface RegisterAddressResponse {
+  success: boolean;
```

Comment:
@mpblocky  That schema doesn't correspond to what we have on BE:
```
{
  "key": "string",
  "value": "string"
}
```
so because of that UI displays schema validation error after successful operation.

If we don't use the response somehow later it might worth to keep it `z.unknown()` or don't validate at all